main.go

```diff
@@ -151,7 +151,11 @@ func visit(url *url.URL) {
 	if (httpMethod == "POST" || httpMethod == "PUT") && postBody == "" {
 		log.Fatal("must supply post body using -d when POST or PUT is used")
 	}
-	req, err := http.NewRequest(httpMethod, url.String(), strings.NewReader(postBody))
+	body, err := createBody(postBody)
```

Comment:
Yes, this is probably the case where it is worth it. I though about it earlier. Done.